src/backend/cpu/kernel/medfilt.hpp

```diff
@@ -97,9 +104,9 @@ void medfilt2(Param<T> out, CParam<T> in, dim_t w_len, dim_t w_wid) {
                     for (int wj = 0; wj < (int)w_wid; ++wj) {
                         bool isColOff = false;
 
-                        int im_col = col + wj - w_wid / 2;
-                        int im_coff;
-                        switch (Pad) {
+                        int im_col  = col + wj - w_wid / 2;
+                        int im_coff = 0;
+                        switch (pad) {
```

Comment:
Should keep this as a template parameter since it in the inner loop of the kernel. This will avoid branching.